File: CTDash/classes/tech.py
# [LATER] — depends on staffing_manager.py
# Tech class — represents a CT technologist.
#
# Status values: "scanning", "paperwork", "idle", "off_shift"

from dataclasses import dataclass, field
from enum import Enum, auto
from typing import Optional


class TechStatus(Enum):
    SCANNING   = auto()
    PAPERWORK  = auto()
    IDLE       = auto()
    OFF_SHIFT  = auto()


@dataclass
class Tech:
    tech_id:        str
    name:           str
    speed:          float           # 0.0–1.0, drives scan time (±25%)
    accuracy:       float           # 0.0–1.0, drives cooldown (±22%) and written-up error rate
    willingness:    float           # 0.0–1.0, drives setup time (±15%) and collaboration floor
    diligence:      float           # 0.0–1.0, drives fireable error probability
    shift_start:    float           # game-time hour, e.g. 7.0
    shift_end:      float           # game-time hour, e.g. 15.5

    assigned_scanner: Optional[str] = None   # scanner_id or None
    status: TechStatus = field(default=TechStatus.OFF_SHIFT)

File: CTDash_Unity/Assets/Scripts/Classes/Tech.cs
// Tech — represents a CT technologist.

namespace CTDash
{
    public enum TechStatus
    {
        Scanning,
        Paperwork,
        Idle,
        OffShift,
    }

    public class Tech
    {
        public string     TechId;
        public string     Name;
        public float      Speed;           // 0.0–1.0, affects scan time modifier
        public float      Accuracy;        // 0.0–1.0, affects error/repeat rate
        public float      Willingness;     // 0.0–1.0, affects refusal to take patients
        public float      KnowledgeBase;   // 0.0–1.0, affects contrast/protocol decisions
        public float      ShiftStart;      // game-time hour, e.g. 7.0
        public float      ShiftEnd;        // game-time hour, e.g. 15.5

        public string     AssignedScanner; // ScannerId or null
        public TechStatus Status = TechStatus.OffShift;

        public Tech(string techId, string name, float speed, float accuracy,
                    float willingness, float knowledgeBase,
                    float shiftStart, float shiftEnd)
        {
            TechId        = techId;
            Name          = name;
            Speed         = speed;
            Accuracy      = accuracy;
            Willingness   = willingness;
            KnowledgeBase = knowledgeBase;
            ShiftStart    = shiftStart;
            ShiftEnd      = shiftEnd;
        }
    }
}
